### ISSUE TITLE
JetEngine 3.1.2 Breaks Grid/Slider Gallery Dynamic Field

### ISSUE THREAD

**Issue by MikeLikesThat** [CrocoBlock/suggestions#6598]
**Describe the bug**
Updating JetEngine from 3.1.1 to 3.1.2 breaks the Grid/Slider Gallery Dynamic Field feature. If the gallery has a list of URLs then it only shows the first image. The first div contains the full URL like <a href="https://...> and <img src="https://...>, but for all additional divs it has blank URLs like <a href class...> and <img src alt...>. The HTML structure is correct for the number of objects in the gallery field but the URLs are not being added.

**To Reproduce**
1. Create Meta Field with the Field Type of Gallery and Value Format of Media URL.
2. Import JSON array of image file URLs
3. In single post type add a Dynamic Field
4. Select the image gallery Meta field as the source
5. Filter field output to be an Images gallery grid

**Expected behavior**
Following the steps above using JetEngine 3.1.1 will produce a grid of images with as many images in the gallery meta field.

**Screenshots**
Here is a video of my setup showing it broken on 3.1.2 and then working once I rollback to 3.1.1: https://www.loom.com/share/6ab5f6e9adca4d76b909c49ddd84f430

**Desktop:**
 - OS: Windows 10
 - Browser: Chrome
 - Version 110.0.5481.178 (Official Build) (64-bit)

**Additional context**
Changing the Meta Field Value Format does not fix it.

**Comment by Floydtm** [comment]
We have checked your request and it seems to work correctly. However, the problem may be on your side. Contact our support team so we can investigate and resolve this issue. Or describe in detail point 2 of your case. You can also add an example json file. Thank you!

**Comment by MikeLikesThat** [comment]
For importing I need recurring imports from a MLS API. I'm currently using WP All Import to achieve this.

Example of the API JSON output:
`  "@odata.context": "https://...",
  "@odata.nextLink": "https://...",
  "@odata.count": 714,
  "value": [
    {
      "@odata.context": "https://...",
      "@odata.id": "https://...",
      "StreetDirPrefix": null,
      "StreetName": "BAYSHORE",
      "LotSizeAcres": 0.25,
      "SubdivisionName": "SUNSET HILLS",
      "Media": [
        {
          "Order": "1",
          "MediaURL": "https://dvvjkgh94f2v6.cloudfront.net/735d922b/594419386/83dcefb7.jpeg",
        },
        {
          "Order": "2",
          "MediaURL": "https://dvvjkgh94f2v6.cloudfront.net/735d922b/594419386/1ad5be0d.jpeg",
        },
        {
          "Order": "3",
          "MediaURL": "https://dvvjkgh94f2v6.cloudfront.net/735d922b/594419386/6dd28e9b.jpeg",
        },...`

Example of how it stores in the DB table wp_postmeta:
meta_key:
image-gallery
meta_value:
https://dvvjkgh94f2v6.cloudfront.net/735d922b/594419386/83dcefb7.jpeg, https://dvvjkgh94f2v6.cloudfront.net/735d922b/594419386/1ad5be0d.jpeg, https://dvvjkgh94f2v6.cloudfront.net/735d922b/594419386/6dd28e9b.jpeg

The value delimiter is a comma + space like this ", "

In the video, in the starter post, you see how it looks in the wp dashboard and also shows in the builder the widget does read the full row of data from the wp_postmeta by loading all the URLs with a ", " delimiter. Updating to 3.1.2 breaks the URL from being added to any "div" and "a" for any object after the first one. Once I rolled back to 3.1.1 it immediately worked with the same data formatting.

In the changelog under 3.1.2 it states:
FIX: Gallery Grid Callback: a warning with URL and Array image value;

Could adding this FIX inadvertently cause this issue?

**Comment by MikeLikesThat** [comment]
This problem still persists with version 3.1.3

**Comment by MikeLikesThat** [comment]
I have it working now but it is still an issue. I found the reason and it is really annoying. At least it happened when developing the site and not a live site.

**Issue:**
As of version 3.1.2 the Meta field type Gallery saves data to that field as "https://url.com/1.jpeg,https://url.com/2.jpeg,etc...".
NOTE: There are NO spaces after the ",".

In version 3.1.1 it saved that data as "https://url.com/1.jpeg, https://url.com/2.jpeg, etc...".
NOTE: There IS a space after the "," making it ", ".

In 3.1.2 this makes all posts created or updated prior to 3.1.2 only show the first image since there is a space between all URLs. 

**To fix it:**
Going into a post created pre 3.1.2 I can simply click to Update the post and it will then save the URL data with no spaces. In turn, it will load all images correctly.

For anyone else who runs into this issue with a lot of posts the best option is to hop into the DB and run a command to clear any spaces from the wp_postmeta tables image-gallery meta_values from your custom post types. Like this:
```
UPDATE wp_postmeta
SET meta_value = REPLACE(meta_value, ' ', '')
WHERE meta_key = 'image-gallery'
```

**Conclusion:**
The main issue is that I have over 10,000 posts created before the 3.1.2 update that needs to be updated for this to be fixed. Luckily as I stated earlier I'm developing the site and when I make it live I'll be grabbing all the posts fresh from an API.

**This was a major data format change and there should have been code to read either format for backward compatibility.**

If any backward compatibility code could be added soon to the next update that would be awesome. This would solve the issue for any live sites not able to update the plugin yet or other sites who updated and have not noticed it yet.
